### PR TITLE
Changes to darray code -  moved some code blocks to internal functions

### PR DESCRIPTION
**Pull request by edhartnett**
Part of #973.

This PR does not change any existing functionality. I moved some of the code blocks in pio_write_darray_multi_nc_serial() to internal helper functions. I also changed the name of the function to write_darray_multi_serial() and cleaned up some of the parameters.

Also added a test (test_darray_multivar2.c) with the test that demonstrates the issue. (Test is commented out so CI won't fail).

This is in preparation for solving the #973 issue for serial writes.

I will merge to develop for testing.

### CODE REVIEW

File: src/clib/pio_darray.c [hunk 1]
```diff
@@ -49,6 +49,27 @@ PIO_Offset PIOc_set_buffer_size_limit(PIO_Offset limit)
...
+ * <li>Allocates a buffer big enough to hold all the data in the
+ * multi-buffer, for all tasks.
+ * <li>Calls rearrange_comp2io() to move data from compute to IO
+ * tasks.
+ * <li>For parallel iotypes (pnetcdf and netCDF-4 parallel) call
+ * pio_write_darray_multi_nc().
+ * <li>For serial iotypes (netcdf classic and netCDF-4 serial) call
+ * pio_write_darray_multi_nc_serial().
+ * <li>For subset rearranger, create holegrid to write missing
+ * data. Then call pio_write_darray_multi_nc() or
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: Changed the name of the call but not here.

File: src/clib/pio_darray.c [hunk 2]
```diff
@@ -283,36 +299,34 @@ int PIOc_write_darray_multi(int ncid, const int *varids, int ioid, int nvars,
...
-
  * <li>Find out how much free space is available in the multi buffer
  * and flush if needed.
- *
  * <li>Store the new user data in the mutli buffer.
- * <li>If needed, fill in gaps in data will fillvalue.
- * <li>
- * <li>
+ * <li>If needed (only for subset rearranger), fill in gaps in data
+ * will fillvalue.
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: with fillvalue

Comment (edhartnett) [inline src/clib/pio_darray.c]: Thanks, will fix.

File: src/clib/pio_darray.c (hunk 1 above)
Comment (edhartnett) [inline src/clib/pio_darray.c]: Thanks, will fix.

**Pull request merged**